app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use Illuminate\Http\Request;
+use App\Models\Category;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\Category\StoreRequest;
+use App\Http\Requests\Category\UpdateRequest;
+use Illuminate\Support\Facades\Redirect;
+
+class CategoryController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categories = Category::orderBy('created_at', 'DESC')
+            ->paginate(config('pagination.per_page'));
+        $parent_cate = Category::with('parent')
+            ->whereNull('parent_id')
+            ->get();
```

Comment:
dòng 21 em đã lấy ra toàn bộ category, dòng 23 em lại lấy ra nữa thành 2 lần truy vấn vào DB.
Nên dòng 23 em dùng lại kết quả của dòng 21 để lấy dữ liệu theo mong muốn, thay with=load nhé